src/components/common/UserProfileEdit/UserProfileEdit.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import { Navigate } from 'react-router-dom';
+
+import { Tab, TabList, TabPanel, Tabs } from '~shared/Tabs';
+
+import ProfileTemplate from '~frame/ProfileTemplate';
+// import LandingPage from '~frame/LandingPage';
+
+import { useAppContext } from '~hooks';
+import { LANDING_PAGE_ROUTE } from '~routes';
+
+import UserProfileSpinner from '../UserProfile/UserProfileSpinner';
+import Sidebar from './Sidebar';
+import UserMainSettings from './UserMainSettings';
+import UserAdvancedSettings from './UserAdvancedSettings';
+
+const displayName = 'common.UserProfileEdit';
+
+const MSG = defineMessages({
+  headingMain: {
+    id: `${displayName}.headingMain`,
+    defaultMessage: 'User profile',
+  },
+  headingAdvance: {
+    id: `${displayName}.headingAdvance`,
+    defaultMessage: 'Advanced settings',
+  },
+});
+
+const UserProfileEdit = () => {
+  const { user, userLoading } = useAppContext();
+
+  if (userLoading || user === undefined) {
+    return <UserProfileSpinner />;
+  }
+
+  if (user === null) {
+    // return <LandingPage />;
```

Comment:
Is this still needed ?

Reply:
It was work in progress as I was trying to fix the loading. Left if as I wanted to discuss it with you. I've updated it all now to the state that I think it should be with the current limitations.

It's still not working perfectly. Either we have landing page showing up while the wallet is connecting or we don't have landing page shown at all if we the wallet is not connected and we just see the loading spinner for a long time and then the `Oh noes! We tried and tried but couldn't get any response from the service.` message.
I've opted for the first option.

Maybe I'm missing something - open to suggestions.